gns3server/handlers/api/controller/node_handler.py

```diff
@@ -59,16 +61,40 @@ class NodeHandler:
     @Route.get(
         r"/projects/{project_id}/nodes",
         parameters={
-            "project_id": "Project UUID"
+            "project_id": "Project UUID",
+            "node_type": "Node type",
+            "status": "Status",
+            "name": "Node name",
+            "locked": "Locked"
         },
         status_codes={
             200: "List of nodes returned",
         },
         description="List nodes of a project")
     async def list_nodes(request, response):
+        project_id = request.match_info["project_id"]
+        node_type = request.query.get("node_type", None)
+        status = request.query.get("status", None)
+        name = request.query.get("name", None)
+        locked = request.query.get("locked", None)
 
-        project = await Controller.instance().get_loaded_project(request.match_info["project_id"])
-        response.json([v for v in project.nodes.values()])
+        project = await Controller.instance().get_loaded_project(project_id)
+        nodes = [v for v in project.nodes.values()]
+
+        if node_type is not None:
+            nodes = [node for node in nodes if node["node_type"] == node_type]
+
+        if status is not None:
+            nodes = [node for node in nodes if "status" in node and node["status"] == "started"]
+
+        if name is not None:
+            nodes = [node for node in nodes if node["name"] == name]
+
+        if locked is not None:
+            locked = locked == "true"
+            nodes = [node for node in nodes if node["locked"] is locked]
```

Comment:
After calling server I get the following error:
TypeError: 'Node' object is not subscriptable
2020-04-03 15:52:50 WARNING crash_report.py:84 A .git directory exist crash report is turn off for developers
2020-04-03 15:52:50 INFO web_log.py:233 ::1 [03/Apr/2020:13:52:50 +0000] "GET /v2/projects/7dc645ff-ddff-40ec-9f72-8447918e5b32/nodes?locked=false HTTP/1.1" 500 935 "-" "PostmanRuntime/7.22.0"

Reply:
Same as above, you should use the node.locked property instead of node["locked"]

Reply:
when I change to node.locked and make request I get

```
Traceback (most recent call last):
  File "C:\Users\piotr.pekala\Desktop\repo\gns3-server\gns3server\web\route.py", line 198, in control_schema
    await func(request, response)
  File "C:\Users\piotr.pekala\Desktop\repo\gns3-server\gns3server\handlers\api\controller\node_handler.py", line 95, in list_nodes
    nodes = [node for node in nodes if node.locked is locked]
  File "C:\Users\piotr.pekala\Desktop\repo\gns3-server\gns3server\handlers\api\controller\node_handler.py", line 95, in <listcomp>
    nodes = [node for node in nodes if node.locked is locked]
AttributeError: 'dict' object has no attribute 'locked'
```

Reply:
Make sure you work on an array of Node objects then you should be able to use the locked property: https://github.com/GNS3/gns3-server/blob/2.3/gns3server/controller/node.py#L240L242

Reply:
Some like this seems to work:

```
        if node_type is not None:
            nodes = [node for node in nodes if node.node_type == node_type]

        if status is not None:
            nodes = [node for node in nodes if node.status == status]

        if name is not None:
            nodes = [node for node in nodes if node.name == name]

        if locked is not None:
            locked = locked == "true"
            nodes = [node for node in nodes if node.locked is locked]
```